src/els_text_synchronization.erl

```diff
@@ -1,22 +1,39 @@
 -module(els_text_synchronization).
 
+-include("erlang_ls.hrl").
+
 -export([ did_open/1
         , did_save/1
         , did_close/1
         ]).
 
+-export([ generate_diagnostics/1 ]).
+
 -spec did_open(map()) -> ok.
 did_open(Params) ->
   TextDocument = maps:get(<<"textDocument">>, Params),
   Uri          = maps:get(<<"uri">>         , TextDocument),
   Text         = maps:get(<<"text">>        , TextDocument),
   Document     = els_dt_document:new(Uri, Text),
-  ok           = els_indexer:index(Document).
+  ok           = els_indexer:index(Document),
+  spawn (?MODULE, generate_diagnostics, [Uri]),
+  ok.
 
 -spec did_save(map()) -> ok.
 did_save(Params) ->
   TextDocument = maps:get(<<"textDocument">>, Params),
   Uri          = maps:get(<<"uri">>         , TextDocument),
+  generate_diagnostics(Uri).
```

Comment:
Sorry I didn't mentioned it before but I think it would make more sense if we also used the spawn here instead of in `els_methods/didsave/2` function.